demos/python_demos/formula_recognition_demo/formula_recognition_demo.py

```diff
@@ -140,90 +203,98 @@ def build_argparser():
-            PREPROCESSING[args.preprocessing_type], image_raw, target_shape)
-        record = dict(img_name=filenm, img=image, formula=None)
-        images_list.append(record)
-
-    log.info("Loading networks")
-    exec_net_encoder = ie.load_network(network=encoder, device_name=args.device)
-    exec_net_decoder = ie.load_network(network=dec_step, device_name=args.device)
-
-    log.info("Starting inference")
-    for rec in tqdm(images_list):
-        image = rec['img']
 
-        enc_res = exec_net_encoder.infer(inputs={args.imgs_layer: image})
-        # get results
-        row_enc_out = enc_res[args.row_enc_out_layer]
-        dec_states_h = enc_res[args.hidden_layer]
-        dec_states_c = enc_res[args.context_layer]
-        output = enc_res[args.init_0_layer]
-
-        tgt = np.array([[START_TOKEN]])
-        logits = []
-        for _ in range(args.max_formula_len):
-            dec_res = exec_net_decoder.infer(inputs={args.row_enc_out_layer: row_enc_out,
-                                                     args.dec_st_c_layer: dec_states_c,
-                                                     args.dec_st_h_layer: dec_states_h,
-                                                     args.output_prev_layer: output,
-                                                     args.tgt_layer: tgt
-                                                     }
-                                             )
-
-            dec_states_h = dec_res[args.dec_st_h_t_layer]
-            dec_states_c = dec_res[args.dec_st_c_t_layer]
-            output = dec_res[args.output_layer]
-            logit = dec_res[args.logit_layer]
-            logits.append(logit)
-            tgt = np.array([[np.argmax(logit, axis=1)]])
-
-            if tgt[0][0][0] == END_TOKEN:
-                break
-        if args.perf_counts:
-            log.info("Encoder performance statistics")
-            print_stats(exec_net_encoder)
-            log.info("Decoder performance statistics")
-            print_stats(exec_net_decoder)
-
-        logits = np.array(logits)
-        logits = logits.squeeze(axis=1)
-        targets = np.argmax(logits, axis=1)
-        if args.output_file:
-            with open(args.output_file, 'a') as output_file:
-                output_file.write(rec['img_name'] + '\t' +  vocab.construct_phrase(targets) + '\n')
+    args = build_argparser().parse_args()
+    interactive_mode = not (os.path.isdir(args.input) or args.input.endswith('.png') or args.input.endswith('.jpg'))
+    args.__dict__['interactive_mode'] = interactive_mode
+    model = Model(args)
+    if not interactive_mode:
+        non_interactive_demo(model, args)
+        return
+
+    *_, height, width = model.encoder.input_info['imgs'].input_data.shape
+    prev_text = ''
+    demo = InteractiveDemo((height, width), resolution=args.resolution)
+    capture = create_capture(args.input, demo.resolution)
+    if not capture.isOpened():
+        log.error("Cannot open camera")
+        return()
```

Comment:
You should exit with non zero code from app.

Reply:
fixed